controllers/idle_handlers.go

```diff
@@ -18,16 +18,49 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"os"
 
 	ranv1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+func getOSName(seedImageVersion string) string {
+	return fmt.Sprintf("rhcos_%s", seedImageVersion)
+}
+
+func (r *ImageBasedUpgradeReconciler) removeDeployment(osName string) {
```

Comment:
Should this be a member function of the ostree client instead?

Reply:
The client is for `rpm-ostree` not `os-tree` . If you prefer it can be moved there though.

Reply:
Ah, ok. I think we will be using various `ostree` commands in multiple handlers. It's good to encapsulate them into another client type wrapper in internal. We can do it in another PR later though.

Reply:
yes, i agree, but I thought this was part of the ostree plugin you mentioned.

Reply:
Actually, should we be using `rpm-ostree cleanup` instead? @donpenney

Reply:
I don't think `rpm-ostree cleanup` is equivalent to `ostree undeploy index`

Reply:
Ok, so we will be using both commands then. That's probably why it's better to have an ostree plugin that encapsulate them.